epuck_ros2_cpp/src/driver.cpp

```diff
@@ -454,29 +543,36 @@ class EPuckPublisher : public rclcpp::Node {
     if (success) {
       publishDistanceData(stamp);
       publishOdometryData(stamp);
+      publishIlluminanceData(stamp);
       mI2cMainErrCnt = 0;
     } else
       mI2cMainErrCnt++;
 
     publishImuData();
+    publishGroundSensorData();
   }
 
   OnSetParametersCallbackHandle::SharedPtr mCallbackHandler;
 
   rclcpp::TimerBase::SharedPtr mTimer;
   rclcpp::Publisher<sensor_msgs::msg::Imu>::SharedPtr mImuPublisher;
+  rclcpp::Publisher<sensor_msgs::msg::Illuminance>::SharedPtr mIlluminancePublisher[8];
   rclcpp::Publisher<sensor_msgs::msg::LaserScan>::SharedPtr mLaserPublisher;
   rclcpp::Publisher<nav_msgs::msg::Odometry>::SharedPtr mOdometryPublisher;
   rclcpp::Publisher<sensor_msgs::msg::Range>::SharedPtr mRangePublisher[8];
+  rclcpp::Publisher<sensor_msgs::msg::Range>::SharedPtr mGroundRangePublisher[3];
   rclcpp::Publisher<sensor_msgs::msg::Range>::SharedPtr mRangeTofPublisher;
-  rclcpp::Subscription<geometry_msgs::msg::Twist>::SharedPtr mSubscription;
+  rclcpp::Subscription<geometry_msgs::msg::Twist>::SharedPtr mTwistSubscription;
+  rclcpp::Subscription<std_msgs::msg::Int32>::SharedPtr mRgbLedSubscription[4];
+  rclcpp::Subscription<std_msgs::msg::Bool>::SharedPtr mLedSubscription[4];
 
   std::unique_ptr<tf2_ros::StaticTransformBroadcaster> mLaserBroadcaster;
   std::unique_ptr<tf2_ros::TransformBroadcaster> mDynamicBroadcaster;
+  std::unique_ptr<tf2_ros::StaticTransformBroadcaster> mGroundBroadcasters[3];
   std::unique_ptr<tf2_ros::StaticTransformBroadcaster> mInfraredBroadcasters[9];
+  std::unique_ptr<tf2_ros::StaticTransformBroadcaster> mLightSensorBroadcasters[8];
```

Comment:
I think it would be nice to introduce `DEFINE` for these numbers as they are repeated at many place in the code, e.g.:
```
#define NB_LIGHT_SENSORS 8
#define NB_INFRARED_SENSORS 9
...
```

Reply:
Fixed. To make more clear I have done something similar with `MSG_SENSORS_*_INDEX` and `MSG_ACTUATORS_*_INDEX`